pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -151,25 +182,8 @@ type KSMConfig struct {
 	// UseAPIServerCache enables the use of the API server cache for the check
 	UseAPIServerCache bool `yaml:"use_apiserver_cache"`
 
-	// CollectPodsFromKubelet enables the collection of pods from the kubelet
-	// instead of the Kubernetes API server.
-	//
-	// This is meant to be enabled when the check is running on the node agent.
-	// This is useful in clusters with a large number of pods where emitting pod
-	// metrics from a single instance might be too much.
-	//
-	// One thing to note is that when the node agent collects metrics from the
-	// kubelet and the cluster agent or cluster check runner collects metrics
-	// for other resources, label joins are not supported for pod metrics if the
-	// join source is not a pod.
-	CollectPodsFromKubelet bool `yaml:"collect_pods_from_kubelet"`
-
-	// CollectOnlyUnassignedPods enables pod collection only for unassigned
-	// pods. This is meant to be enabled when the check is running on the
-	// cluster agent or the cluster check runner and "CollectPodsFromKubelet" is
-	// enabled on the node agents, because unassigned pods cannot be collected
-	// from node agents.
-	CollectOnlyUnassignedPods bool `yaml:"collect_only_unassigned_pods"`
+	// PodCollectionMode defines how pods are collected.
+	PodCollectionMode podCollectionMode `yaml:"pod_collection_mode"`
```

Comment:
I don't know how godoc si generating the doc with the kind of "enum" type description.
but it can be good to add a comment here with the list of possible value so it is easier to read.

Reply:
Done :+1: